lib/rules/no-irregular-whitespace.js

```diff
@@ -65,6 +62,7 @@ module.exports = {
         const skipTemplates = !!options.skipTemplates;
 
         const sourceCode = context.getSourceCode();
+        const commentNodes = sourceCode.getAllComments();
```

Comment:
This rule was using comment order traversal to collect comment tokens, but didn't need to. Its goal is to check all comments, so instead of collecting them while traversing we can simply use `sourceCode.getAllComments()`.